Remove usage of execSync in scripts

skoeva opened the pull request:
These changes replace `execSync` with `execFileSync` in the two places where
dynamic values are interpolated into a shell command string.

- `app/electron/main.ts`: `list-plugins` built its command from `backendPath`,
  which comes from `process.resourcesPath`. An install path with a space or a
  shell metacharacter would break the command or run arbitrary shell.
- `tools/releaser/src/utils/git.ts`: `version` and `branchName` come from
  releaser CLI arguments, and `git add` interpolated filesystem paths.

Note: the `git` calls in the releaser now use `stdio: 'inherit'` where they
previously defaulted to `'pipe'`, so their output is visible.

joaquimrocha commented on plugins/headlamp-plugin/bin/headlamp-plugin.js:
```diff
@@ -1024,15 +1023,23 @@ function upgrade(packageFolder, skipPackageUpdates, headlampPluginVersion) {
-        encoding: 'utf8',
-      });
-    } catch (e) {
+    const result = child_process.spawnSync(cmd, {
```
Curious, why is this one not execFileSync? (assuming you can split cmd?)
Not saying it's better, just curious.

illume commented on plugins/headlamp-plugin/bin/headlamp-plugin.js:
```diff
@@ -1260,16 +1267,16 @@ function tsc(packageFolder) {
-  } catch (e) {
+  const command =
+    './node_modules/.bin/storybook dev -p 6007 -c node_modules/@kinvolk/headlamp-plugin/config/.storybook';
+  const result = child_process.spawnSync(command, {
```
looks like this was done

illume reviewed (changes requested):
I think these require testing. The copilot found one issue which turned out to be a real one, so I'm worried that these changes will break other things.

It might be worth considering which if any of these is a security problem really to reduce the amount of testing that needs to be done.

skoeva committed:
app: electron: Replace execSync with execFileSync

The list-plugins command interpolated backendPath, which is derived from
process.resourcesPath, into a shell command string. An install path
containing a space or a shell metacharacter would break the command or
allow arbitrary shell execution. taskkill is switched over too so the
file no longer needs execSync.

Signed-off-by: Evangelos Skopelitis <eskopelitis@microsoft.com>

skoeva committed:
tools/releaser: git: Replace execSync with execFileSync

The version and branchName values come from releaser CLI arguments and
were interpolated straight into shell command strings, and 'git add'
interpolated filesystem paths. Passing arguments as an array removes the
shell from the picture.

Signed-off-by: Evangelos Skopelitis <eskopelitis@microsoft.com>

illume reviewed (approved):
🎉 thanks!
